command_line/generate_mask.py

```diff
@@ -85,55 +86,70 @@
 def generate_mask(experiments, params):
     # type: (ExperimentList, phil.scope_extract) -> Tuple[Masks, Optional[ExperimentList]]
     """
-    Generate a pixel mask for each image in an experiment.
+    Generate a pixel mask for each imageset in an experiment list.
 
-    Use the masking parameters :param:`params` and an experiment in the experiment
-    list :param:`experiments` to define pixel masks for the associated imageset.
-    The masks are generated using :mod:`dials.util.masking`.
+    Use the masking parameters :param:`params` and the experiments in the experiment
+    list :param:`experiments` to define a pixel mask for each of the associated
+    imagesets.  The masks are generated using :mod:`dials.util.masking`.
 
     The masks will be saved to disk at the location specified by
-    :attr:`params.output.mask`.  Optionally, if a path
-    :attr:`params.output.experiments` is set, a modified copy of :param:`experiments`
-    with the masks applied will be saved to that location.
+    :attr:`params.output.mask`.  If the experiment list contains more than one
+    imageset, multiple mask files will be produced, with filenames differentiated by
+    an appended number.  Optionally, if a path :attr:`params.output.experiments` is
+    set, a modified copy of :param:`experiments` with the masks applied will be saved
+    to that location.
 
     Args:
         experiments: An experiment list containing only one imageset.
         params: Masking parameters, having the structure defined in
             :data:`phil_scope`.
 
     Returns:
-        A tuple containing the generated pixel masks.
+        A list of masks, one for each imageset.
 
         A copy of :param:`experiments` with the masks applied (optional,
         only returned if :attr:`params.output.experiments` is set).
     """
     imagesets = experiments.imagesets()
-    assert len(imagesets) == 1, "Only single imagesets supported"
-
-    imageset = imagesets[0]
+    masks = []
+
+    # Specify the formatting of the output filenames
+    num_imagesets = len(imagesets)
+    pad = len("{:d}".format(num_imagesets))
```

Comment:
`pad = len(str(num_imagesets))` a bit clearer? I don't think the `:d` actually buys you anything here?

Reply:
Also this is a more picky comment but I'm not sure that `num_imagesets = len(imagesets)` is saving much - only one less character for the extra indirection.

Reply:
Good point, thanks.  I'll push a change when you're done reviewing.